### ISSUE TITLE
file upload

### ISSUE THREAD

**Issue by angelistudio** [psztrnk/catalog2#1]
Hi there, 
first of all I really enjoy your software, I'm an avid book reader and I was looking for something like this. I always wanted to build something myself, but I think catalog2 does wonderfully the job.

I installed everything on my raspberry pi 3. It works like a charm but I'm having trouble uploading ebooks: it simply doesn't load the file...

maybe it's some obscure configuration on my server?

Ciao.

Marco

**Comment by psztrnk** [comment]
Hi Marco,

Thanks for the feedback.

Can you please clarify this a bit: "simply doesn't load the file"?

I wonder whether the upload doesn't work, or the upload works and you can't open/download the uploaded file?

Thanks,
Adam

**Comment by angelistudio** [comment]
Hey Adam,
sorry to bug you this saturday night...
anyway: the upload doesn't work for me. The "file upload" pop up appears, I choose a pdf but nothing happens: no file selected, no sign of life. Saving the new entry doesn't make a difference. 

Just checked: my ebook folder is 755.

Cheers from windy Cividale in Italy.

Marco

**Comment by psztrnk** [comment]
Hey Marco,

Thanks for clarifying.
Just checked on a fresh install on my live server, trying to reproduce the issue -- file upload (tried with PDF, mobi and txt files) works OK for me (even on mobile).

Please hang on with this for a while longer: I have a spare rasPi somewhere, so I'll also do an install on it to see whether the issue pops up on that environment.

Thanks,
Adam

**Comment by psztrnk** [comment]
Hi Marco,

sorry this took me so long. 
I've been trying to reproduce the issue on my RasPi and could only do it one way: by altering the file upload related values in my php.ini.

Can you please check/try the following:

1. open up your php.ini file in your text editor (usually located at etc/php/[version]/apache2/php.ini)
2. go to the "File Uploads" section (should be around the middle of the file)
3. "file_uploads" value should be = "On"
4. "upload_max_filesize" should be something bigger than the default, I use "128M" (so you can upload bigger files)

Save the file and don't forget to restart the apache2 process so it can pick up the changes you've made.

Thanks,
Adam

**Comment by psztrnk** [comment]
Oh, one more thing: when you browse the file to be uploaded, the filename will not be displayed after selecting it. This is known Bootstrap thing: the styling masks the field text. So when you select the file, it is not visible, but it's there. :)

**Comment by angelistudio** [comment]
well Adam,
after a year I solved my issue now.

I just installed this software, https://automad.org/, but I couldn't update any page. So I just gave "777" file permissions to all the folders (with recurse to all files and sub-directories) and automad worked like a charm.

Then I tought: I have catalogue 2, let's try the same trick...

...it was a permission issue on my raspberry pi 3 server.

Thanks again for your kind support, have a nice day!

Ciao

Marco
